### ISSUE TITLE
Force stdlib  etree instead of lxml?

### ISSUE THREAD

**Issue by narfel** [tkrajina/gpxpy#248]
For my use case, when lxml is installed I get drastically less performance. In the __init__.py it says "

> parser may be 'lxml', 'minidom' or None (then it will be automatically  detected, lxml if possible).

How do I tell gpxpy to use that? parse(xml_or_file: Union[AnyStr, IO[str]], version: Optional[str] = None) has no option for a parser, or am I missing something?

**Comment by tkrajina** [comment]
No, you're right. When I first implemented this lxml was faster, so I made to be default. But it absolutely makes sense to use the default parser by default. Maybe something like this: https://github.com/tkrajina/gpxpy/commit/b75314df87b9fd5e7e267bbfc13159624da16533

That way, you gahe a global `gpx.set_xml_etree()` function where you can set lxml if you want.

**Comment by thomthom** [comment]
I'm observing that `gpxpy` is faster without `lxml`. Currently using Python 3.11. Maybe the stdlib has gotten faster?

I have been testing with a bunch of GPX files I have, total 5500km of tracks, collected from my GPX while riding my motorcycle.

I did some simple benchmarking of computing the total distance of all the GPX files I had, using `gpxpy`, `xml.etree` and `lxml`:

```
Running 3 benchmarks with 3 iterations...

Running xml_etree ...
xml_etree 5463043.740615641 meters
xml_etree 5463043.740615641 meters
xml_etree 5463043.740615641 meters
xml_etree: 2.342165 seconds (Average: 0.780722 seconds)

Running lxml ...
lxml 5463043.740615641 meters
lxml 5463043.740615641 meters
lxml 5463043.740615641 meters
lxml: 3.194700 seconds (Average: 1.064900 seconds)

Running gpxpy ...
gpxpy 5463041.784135511 meters
gpxpy 5463041.784135511 meters
gpxpy 5463041.784135511 meters
gpxpy: 36.933371 seconds (Average: 12.311124 seconds)
```

I also have benchmarks of `gpxpy` vs `xml.etree` without `lxml`:

```
Running 2 benchmarks with 3 iterations...

Running gpxpy ...
gpxpy 5463041.784135511 meters
gpxpy 5463041.784135511 meters
gpxpy 5463041.784135511 meters
gpxpy: 11.497863 seconds (Average: 3.832621 seconds)

Running xml_etree ...
xml_etree 5463043.740615641 meters
xml_etree 5463043.740615641 meters
xml_etree 5463043.740615641 meters
xml_etree: 2.538976 seconds (Average: 0.846325 seconds)
```

Notice how it made `gpxpy` 7 seconds slower.

I've yet to dive into profiling the code. On my list when I have time. 

But it would certainly be good to control which XML parser `gpxpy` uses in case you end up with a project where one of the dependencies pulls in `xlml` outside of your control. From what I'm seeing the stdlib xml lib is significantly faster.

**Comment by cincodenada** [comment]
I'm also noticing this, for my data gpxpy is twice as slow(!) with lxml installed vs without, which is the opposite of what the docs suggest. I'm using Python 3.10. Being able to choose the parser would be great, but a doc update would be helpful too, I think!

**Comment by tkrajina** [comment]
Yes, lxml is nowadays slower. At the time when I first implemented this it was waay faster.

I'm working on completely removing lxml in https://github.com/tkrajina/gpxpy/tree/rm-lxml . If somebody wants to test it, I'll merge it and make it part of the next release.

**Comment by tkrajina** [comment]
Merged.
